internal/api/web/views/components/quiz_components/answers.templ

```diff
@@ -0,0 +1,79 @@
+package quiz_components
+
+import "github.com/Molnes/Nyhetsjeger/internal/data/questions"
+
+type Optional struct {
+	correctAnswered []string
+}
+
+type Option func(*Optional) Optional
+
+//Sets the correct and answered answer to the question
+func CorrectAndAnswered(correct string, answered string) Option {
+	return func(o *Optional) Optional {
+		o.correctAnswered = append(o.correctAnswered, correct, answered)
+		return *o
+	}
+}
+
+var answeredCorrectly bool = false
+
+//Checks if the answer has been selected
+func isSelected(id string, option ...Option) bool {
+	o := Optional{}
+	for _, opt := range option {
+		o = opt(&o)
+	}
+
+	for i := 0; i < len(o.correctAnswered); i += 2 {
+		if o.correctAnswered[i] == id && o.correctAnswered[i+1] == id {
+			answeredCorrectly = true
+			return true
+		}
+
+		if o.correctAnswered[i+1] == id {
+			return true
+		}
+	}
+
+	return false
+
+}
+
+//Checks if the question has been answered correctly or not.
+func isCorrect(id string, option ...Option) bool {
+	o := Optional{}
+	for _, opt := range option {
+		o = opt(&o)
+	}
+
+	for i := 0; i < len(o.correctAnswered); i += 2 {
+		if o.correctAnswered[i] == id {
+			return true
+		}
+	}
+
+	return false
+}
+
+//Renders all the answers to the question in the form of buttons, and if the question has been answered, it will also display if the answer was correct or not.
+//Takes in the possible questions, and an optional Option to check if the question has been answered and if it was answered correctly.
+templ Answers(answers []questions.Alternative, option ...Option) {
+	<div id="answers-wrapper" class="w-80 flex flex-col [&>*]:mt-4">
+		if (option != nil) {
```

Comment:
again, might be simpler to have a separate component for feedback buttons

Reply:
has been answered